test/rules/no-unused-modules.spec.ts

```diff
@@ -1,15 +1,17 @@
 import fs from 'node:fs'
 
 import { TSESLint } from '@typescript-eslint/utils'
-// @ts-expect-error - no typings yet
-import { FlatRuleTester } from 'eslint/use-at-your-own-risk'
+import { version } from 'eslint/package.json'
+import semver from 'semver'
 
 import { test, testFilePath, parsers } from '../utils'
 
 import jsxConfig from 'eslint-plugin-import-x/config/react'
 import typescriptConfig from 'eslint-plugin-import-x/config/typescript'
 import rule from 'eslint-plugin-import-x/rules/no-unused-modules'
 
+const FlatRuleTester = semver.satisfies(version, '8') ? require('eslint/use-at-your-own-risk').FlatRuleTester : require('eslint').RuleTester
```

Comment:
@JoshuaKGoldberg We want to test this specific rule under both flat rule tester and legacy rule tester. And we want the `eslint-plugin-import-x` to be compatible with both ESLint 8.56 and 9.x. Would you like to share your thoughts on how to do this?

Reply:
Good question... Both ESLint v9 (to my knowledge) and typescript-eslint v8 (as of today) have fully dropped the old rule tester. So there's no way that I know to get legacy config based testing under the current versions.

Maybe you could install both, like having `"eslint8": "eslint@8.57.0"` in `package.json` `devDependencies` or something like that? I haven't tried that out, just a guess.

You might want to file issue(s) on ESLint and/or typescript-eslint to also export a class like `LegacyRuleTester`. If this ends up being a common ecosystem pain, at least in typescript-eslint we'd want to know & have a place for people to post.

Reply:
> Maybe you could install both, like having `"eslint8": "eslint@8.57.0"` in `package.json` `devDependencies` or something like that?

That's a good idea! I could try that out before submitting issues.